reco_utils/recommender/deeprec/models/base_model.py

```diff
@@ -14,7 +12,7 @@
 
 
 class BaseModel(object):
-    def __init__(self, hparams, iterator_creator, graph=None, seed=42):
+    def __init__(self, hparams, iterator_creator, graph=None, seed=None):
```

Comment:
why `None` here? in the rest of the notebooks we have 42

Reply:
The default `seed` should be the same as the functions' that we use in the module, which is `None`. See, e.g., [tensorflow](https://www.tensorflow.org/api_docs/python/tf/random/normal).

This follows the rational that if you don't pass the seed explicitly, the behavior should be 'random' (produce different results every time you run it).
e.g. Python's random behavior: 
> random.seed(a=None, version=2): If a is omitted or None, the current system time is used.

Reply:
if this is the defualt behavior we want to follow, then we should have the same idea in all the other algos, which should be changed. Do we want to follow this idea instead of fixing the seed? @anargyri @yueguoguo @gramhagen

Reply:
I think it is a good practice to use `seed=None` for algo modules, and set seed explicitly from example notebooks.

Reply:
This sounds reasonable, i guess right now you would have to set the seed to None to actually get random values, which may be unexpected. I think we should adopt this as it matches approaches from other libraries and create a separate ticket to change the notebooks / code / tests as needed.

Reply:
in numpy the default is None: https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.seed.html#numpy.random.seed, so I think it makes sense to change all to None

@loomlike do you want to homogenize None in all the utilities and then set 42 explicitly in the notebooks? Not sure if on this PR or in a different want

Reply:
@miguelgfierro I will create a separate issue to handle them